samples/sensor/qdec/sample.yaml

```diff
@@ -0,0 +1,16 @@
+sample:
+  description: Usage quadrature decoder sensor
+  name: qdec_sensor
+tests:
+  sample.sensor.qdec_sensor:
+    tags: sensors
+    filter: dt_alias_exists("qdec0")
```

Comment:
FYI, this won't work to build sample on nucleo_f401re as overlays are applied after filtering.
This could be useful for your own CI set up with a board that defines this alias as default configuration locally.
But in current state of the tree with no board defining it; result is that the sample will never be built.
Besides, I don't encourage adding this alias directly as part of nucleo_f401re default description as the required associated encoder hw is not part of the BSP, so it would not make much sense.

So, I'd suggest to keep `platform_allow: nucleo_f401re`.

Reply:
Thanks! I didn't know that overlays are applied only after filtering. I'll fix this as you suggested